components/support/sync15-traits/src/server_timestamp.rs

```diff
@@ -87,11 +87,23 @@ impl<'de> serde::de::Visitor<'de> for TimestampVisitor {
     fn visit_f64<E: serde::de::Error>(self, value: f64) -> Result<Self::Value, E> {
         Ok(ServerTimestamp::from_float_seconds(value))
     }
+
+    fn visit_i64<E: serde::de::Error>(self, value: i64) -> Result<Self::Value, E> {
+        Ok(if value < 0 {
+            ServerTimestamp(0)
+        } else {
+            ServerTimestamp(value * 1000)
+        })
+    }
+
+    fn visit_u64<E: serde::de::Error>(self, value: u64) -> Result<Self::Value, E> {
+        Ok(ServerTimestamp(value as i64 * 1000))
```

Comment:
Hmm... Should we be using `checked_mul` and `i64::try_from` for these? If we're seeing weird values it might be worth it...